### PR TITLE
Porting Kafka sink to the new api 

### PR DESCRIPTION
**Pull request by huangyuqi**
Porting Kafka sink to the new api

### CODE REVIEW

**Comment by huangyuqi** [comment]
Hi, @mwielgus 
Please help to review, thanks so much.

**Comment by mwielgus** [comment]
Please rebase the PR. Otherwise LGTM.

**Comment by mwielgus** [comment]
Something went wrong with the rebase. There are 3 commits that should not belong to this PR.

**Comment by huangyuqi** [comment]
Sorry, I am checking this issue.

File: metrics/sinks/factory.go [hunk 1]
```diff
@@ -44,7 +45,9 @@ func (this *SinkFactory) Build(uri manager.Uri) (core.DataSink, error) {
 			core.MetricCpuUsageRate.MetricDescriptor.Name,
 			core.MetricMemoruUsage.MetricDescriptor.Name}), nil
 	case "riemann":
-		return riemann.CreateRiemannSink(&uri.Val)
+	    return riemann.CreateRiemannSink(&uri.Val)
```
Comment (mwielgus) [inline metrics/sinks/factory.go]: Please fix the indentation.

**Comment by huangyuqi** [comment]
sorry, @mwielgus , some misoperation occur in rebase. Please merge this PR now, thanks so much.
BTW, would you please share me the task list of heapster reconsitution? ^_^

**Comment by mwielgus** [comment]
lgtm

**Comment by mwielgus** [comment]
Thank you for porting Kafka and Riemann.

**Comment by mwielgus** [comment]
@huangyuqi  I created couple issues and cc-ed you on them (#826, #827, #828). Feel free to pick the one you like (just add a comment if you want to start working on it).

**Pull request merged**